lib/publiccloud/utils.pm

```diff
@@ -49,6 +50,50 @@ sub select_host_console {
     set_var('TUNNELED', $tunneled) if $tunneled;
 }
 
+sub register_addon {
+    my ($remote, $addon) = @_;
+    my $arch = get_var('PUBLIC_CLOUD_ARCH') // "x86_64";
+    $arch = "aarch64" if ($arch eq "arm64");
+    record_info($addon, "Going to register '$addon' addon");
+    if ($addon =~ /ltss/) {
+        ssh_add_suseconnect_product($remote, get_addon_fullname($addon), '${VERSION_ID}', $arch, "-r " . get_required_var('SCC_REGCODE_LTSS'));
+    } elsif (is_sle('<15') && $addon =~ /tcm|wsm|contm|asmm|pcm/) {
+        ssh_add_suseconnect_product($remote, get_addon_fullname($addon), '`echo ${VERSION} | cut -d- -f1`', $arch);
+    } elsif (is_sle('<15') && $addon =~ /sdk|we/) {
+        ssh_add_suseconnect_product($remote, get_addon_fullname($addon), '${VERSION_ID}', $arch);
+    } else {
+        ssh_add_suseconnect_product($remote, get_addon_fullname($addon), undef, $arch);
+    }
+}
+
+sub registercloudguest {
```

Comment:
I would put these function inside of `publiccloud::instance` but up to you.

Reply:
generally I agree but there is strong time pressure with solving this , so I simply don't have time to proceed sorry :(